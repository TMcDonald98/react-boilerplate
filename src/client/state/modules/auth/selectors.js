
export const getApiKey = state => state?.auth?.apiKey;
