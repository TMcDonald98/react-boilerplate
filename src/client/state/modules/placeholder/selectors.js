
export const getPlaceholder = state => state?.placeholder?.data;
export const getApiKey = state => state?.app?.apiKey;
