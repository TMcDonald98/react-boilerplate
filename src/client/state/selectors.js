
export * from './modules/auth/selectors';
export * from './modules/location/selectors';
export * from './modules/placeholder/selectors';
