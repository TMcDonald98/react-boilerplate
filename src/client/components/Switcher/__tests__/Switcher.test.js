import * as React from 'react';
import { getLocationType } from 'state/modules/location/selectors';
import {
  Switcher,
  mapStateToProps
} from '../Switcher';

jest.mock('state/modules/location/selectors');

const testRoute = 'a long time ago';
const testString = 'in a galaxy far far away';
const testProps = {
  routes: {
    [testRoute]: () => testString
  },
  locationType: testRoute
};

describe('components/Switcher', () => {
  describe('render', () => {
    it('should render without error', () => {
      const result = shallow(<Switcher {...testProps} />);

      expect(result).toMatchSnapshot('Switcher');
    });
  });

  describe('mapStateToProps', () => {
    it('should call the right selectors', () => {
      const testState = 'hi';
      mapStateToProps(testState);

      expect(getLocationType).toHaveBeenCalledWith(testState);
    });
  });
});
