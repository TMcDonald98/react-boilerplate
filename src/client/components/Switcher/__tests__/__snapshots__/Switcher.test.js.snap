// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Switcher render should render without error: Switcher 1`] = `<a long time ago />`;
